src/libraries/System.Net.Http/src/System/Net/Http/Headers/ViaHeaderValue.cs

```diff
@@ -281,7 +281,7 @@ private static void CheckReceivedBy(string receivedBy)
 
             // 'receivedBy' can either be a host or a token. Since a token is a valid host, we only verify if the value
             // is a valid host.;
-            if (HttpRuleParser.GetHostLength(receivedBy, 0, true, out string? host) != receivedBy.Length)
+            if (HttpRuleParser.GetHostLength(receivedBy, 0, true, out _) != receivedBy.Length)
```

Comment:
@geoffkizer, not sure how much it actually matters, but looking at the call sites to GetHostLength, it seems several of them end up ignoring the out string host even though the method will end up allocating one, regardless.  Might be worth a look, and if there are other similar cases that could be improved.

Reply:
Yeah, this is ugly. The usual pattern of the GetXLength methods here and elsewhere is that they return an int to indicate the length, but don't actually construct the substring, which the caller can do trivially. Not sure why this one is different. I'll take a look.

Reply:
Fixed in https://github.com/dotnet/runtime/pull/63641
See also https://github.com/dotnet/runtime/issues/63640